packages/@yoda/flora/index.js

```diff
@@ -2,259 +2,161 @@
+ * post msg
  * @method post
- * @param {String} name - msg name to post
- * @param {module:@yoda/flora~Caps} msg - msg content
- * @param {Number} type - msg type to post (exports.MSGTYPE_INSTANT | exports.MSGTYPE_PERSIST)
+ * @memberof module:@yoda/flora~Agent
+ * @param {String} name - msg name
+ * @param {Array} msg - msg content
+ * @param {Number} type - msg type (MSGTYPE_INSTANT | MSGTYPE_PERSIST}
+ * @returns {Number} 0 for success, otherwise error code
  */
-// Client.prototype.prototype.post
 
 /**
- * close flora client
- * @memberof module:@yoda/flora~Client
- * @method close
+ * @class module:@yoda/flora~Response
+ * @classdesc Response of Agent.get returns
  */
-// Client.prototype.prototype.close
-
-Client.prototype.native_callback = function (type, args) {
-  switch (type) {
-    // recv post
-    // params: name msgtype msgcontent
-    case 0:
-      this.callbacks[0](args[0], args[1], args[2])
-      break
-    // disconnected
-    case 1:
-      this.callbacks[1]()
-      break
-  }
-}
 
 /**
- * @memberof module:@yoda/flora
- * @member {Number} MSGTYPE_INSTANT
+ * @memberof module:@yoda/flora~Response
+ * @member {Number} retCode
  */
-exports.MSGTYPE_INSTANT = 0
+
 /**
- * @memberof module:@yoda/flora
- * @member {Number} MSGTYPE_PERSIST
+ * @memberof module:@yoda/flora~Response
+ * @member {Array} msg
  */
-exports.MSGTYPE_PERSIST = 1
-// exports.MSGTYPE_REQUEST = 2;
 
-exports.CLI_SUCCESS = 0
-// 认证失败
-exports.CLI_EAUTH = -1
-// 调用参数错误
-exports.CLI_EINVAL = -2
-// 连接断开
-exports.CLI_ECONN = -3
-// 'get'请求超时
-// exports.CLI_ETIMEOUT = -4;
-// 'get'请求目标不存在
-// exports.CLI_ENEXISTS = -5;
+/**
+ * @memberof module:@yoda/flora~Response
+ * @member {String} sender
+ */
 
 /**
- * @class
- * @classdesc utility of data struct serialize
+ * @callback module:@yoda/flora~SubscribeCallback
+ * @param {Array} - msg content
+ * @param {Number} - type of msg
+ * @returns {module:@yoda/flora~Reply} reply message to sender of this REQUEST message
  */
-function Caps () {
-  this.pairs = []
-}
 
-Caps.INT32 = 105 // 'i'
-Caps.INT64 = 108 // 'l'
-Caps.FLOAT = 102 // 'f'
-Caps.DOUBLE = 100 // 'd'
-Caps.STRING = 83 // 'S'
-Caps.BINARY = 66 // 'B'
-Caps.OBJECT = 79 // 'O'
+var Agent = require('./flora-cli.node').Agent
 
 /**
- * write int32 value
- * @memberof module:@yoda/flora~Caps
- * @method writeInt32
- * @param {Number} i32 - int32 value
- */
-Caps.prototype.writeInt32 = function (i32) {
-  if (typeof i32 !== 'number') { return }
-  var p = { type: Caps.INT32, value: i32 }
-  this.pairs.push(p)
+ * subscribe flora msg
+ * @method subscribe
+ * @memberof module:@yoda/flora~Agent
+ * @param {String} name - msg name for subscribe
+ * @param {module:@yoda/flora~SubscribeCallback} cb - callback if received msg that subscribed
+ */
+Agent.prototype.subscribe = function (name, cb) {
+  this.nativeSubscribe(name, (msg, type) => {
+    try {
+      return cb(msg, type)
+    } catch (e) {
+      process.nextTick(() => {
+        throw e
+      })
```

Comment:
In Node.js, the callback uses the first argument as `error` slot, and throwing an error at `nextTick` callback seems weird and cannot be caught. So my suggestion is:

```js
try {
  return cb(null, msg, type)
} catch (e) {
  process.nextTick(() => cb(e))
}
```

Reply:
throw exception here for oh-my-little-pony catch and print message of exception when exception throw from 'cb' method.
your suggestion is not solve problem.

Reply:
`@yoda/oh-my-little-pony` seems not a runtime dependence of flora client, and Node.js has the callback-style, which requires the callback's 1st argument should be an error. @mingzc0508 may I ask you to describe the problem here that you try to resolve?